synapse/cortex.py

```diff
@@ -904,8 +891,38 @@ async def finidmon():
 
         await self.postNexsAnit()
 
-        if mirror is not None:
-            await self.initCoreMirror(mirror)
+        if self.mirror is not None:
+            await self._initCoreMirror(self.mirror)
+
+        # Enable leadership change awareness and fire
+        # the leadership hook once at boot
+        self.isleader = self.nexsroot.amLeader()
+        await self._initStormSvcs()
+        self.leaderchangeaware = True
+        await self.onLeaderChange(self.isleader)
```

Comment:
only the onLeaderChange API should set self.isleader so i would tweak this to
`self.onLeaderChange(self.nexsroot.amLeader())`